xmppserver/src/main/java/org/jivesoftware/openfire/ldap/LdapManager.java

```diff
@@ -1294,20 +1294,26 @@ private Boolean lookupExistence(InitialDirContext ctx, LdapName dn, String[] ret
      * @param baseDN the base dn to use in the URL.
      * @return the properly encoded URL for use in as PROVIDER_URL.
      */
-    private String getProviderURL(LdapName baseDN) {
+    String getProviderURL(LdapName baseDN) throws NamingException
+    {
         StringBuffer ldapURL = new StringBuffer();
 
-        for (String host : hosts) {
-            // Create a correctly-encoded ldap URL for the PROVIDER_URL
-            ldapURL.append("ldap://");
-            ldapURL.append(host);
-            ldapURL.append(':');
-            ldapURL.append(port);
-            ldapURL.append('/');
-            ldapURL.append(baseDN);
-            ldapURL.append(' ');
-        }
-        return ldapURL.toString();
+        try
+        {
+            for ( String host : hosts )
+            {
+                // Create a correctly-encoded ldap URL for the PROVIDER_URL
+                final URI uri = new URI("ldap", null, host, port, "/" + baseDN.toString(), null, null);
```

Comment:
Perhaps 

`sslEnabled ? "ldaps" : "ldap"`

?? (That said, I've got sslEnabled set and it's working fine so I'm guessing the schema isn't used in practice, but will save problems if/when it does start getting used).

Reply:
Yeah, I noticed that too. Want to tackle that in a separate issue. @speedy01 might be interested in this too.